x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/history/HistoryStore.java

```diff
@@ -78,9 +76,12 @@ public void forcePut(WatchRecord watchRecord) {
      * @return true, if history store is ready to be started
      */
     public static boolean validate(ClusterState state) {
-        String currentIndex = HistoryStoreField.getHistoryIndexNameForTime(ZonedDateTime.now(ZoneOffset.UTC));
-        IndexMetadata indexMetadata = WatchStoreUtils.getConcreteIndex(currentIndex, state.metadata());
+        IndexMetadata indexMetadata = WatchStoreUtils.getConcreteIndex(HistoryStoreField.DATA_STREAM, state.metadata());
         return indexMetadata == null || (indexMetadata.getState() == IndexMetadata.State.OPEN &&
             state.routingTable().index(indexMetadata.getIndex()).allPrimaryShardsActive());
     }
+
+    public void flush() {
+        bulkProcessor.flush();
+    }
```

Comment:
Out of curiosity, is this addressing a scenario that's specific to data streams or just a general case where history entries weren't being written?

Reply:
Main reason for this is to have clear state after tests. This makes sure that we don't have any pending docs to index so we can do proper cleanup after the test and leave no hanging indices behind